cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_custom_slurm_settings_include_file_generator.py

```diff
@@ -0,0 +1,169 @@
+        if key.lower() in ["nodename", "partitionname", "nodeset", "downnodes"]:
+            first_subparam = {key: value}
+            break
+
+    remaining_subparams = {key: value for key, value in param.items() if key not in first_subparam}
+
+    output_string = ""
+    # The first loop will either loop over a single parameter or not run at all
+    for key, value in first_subparam.items():
+        output_string = f"{key}={value}"
+    for key, value in remaining_subparams.items():
+        output_string = " ".join([output_string, f"{key}={value}"])
+
+    return output_string
+
+
+def _generate_include_file(include_target: str, include_file_config: List[Dict], output_directory, dryrun: bool):
+    """
+    Generate single custom Slurm settings include file at a given level.
+
+    :param include_target: target of the custom Slurm setting include file (e.g. slurm, slurmdbd, cgroup, etc.)
+    :param include_file_config: dictionary with relevant custom Slurm settings
+    :param output_directory: output directory where the include file is written
+    :param dryrun: dryrun
+    :return:
+    """
+    log.info("Generating custom_slurm_settings_include_file_%s.conf", include_target)
+    if not dryrun:
+        filename = path.join(output_directory, f"custom_slurm_settings_include_file_{include_target}.conf")
+
+        # The include_file_config is structured as a list of dictionaries, which can either contain one
+        # single key-value pair (simple parameters) or multiple pairs (e.g. partition definitions).
+        with open(filename, "w", encoding="utf-8") as output_file:
+            output_file.write(INCLUDE_FILE_HEADER + "\n\n")
+
+            for param in include_file_config:
+                output_string = (
+                    _generate_simple_parameter(param) if len(param) == 1 else _generate_complex_parameter(param)
+                )
+                output_file.write(output_string + "\n")
+
+
+def _generate_custom_slurm_config_include_files(
+    output_directory,
+    input_file,
+    dryrun,
+):
+    """Generate Custom Slurm Settings include files for the Slurm configuration."""
+    # Make output directories
+    output_directory = path.abspath(output_directory)
+    pcluster_subdirectory = path.join(output_directory, "pcluster")
+    makedirs(pcluster_subdirectory, exist_ok=True)
+
+    # Load cluster configuration from YAML
+    cluster_config = _load_cluster_config(input_file)
+
+    for include_target in ["slurm"]:
+        # Get specific custom Slurm settings section and if present, create the corresponding include file
+        include_file_config = cluster_config["Scheduling"]["SlurmSettings"].get(
+            INCLUDE_TARGET_MAPPING[include_target], {}
+        )
+        if include_file_config:
+            _generate_include_file(
+                include_target,
+                include_file_config,
+                pcluster_subdirectory,
+                dryrun,
+            )
+
+    log.info("Finished.")
```

Comment:
It's inherited from the script I used as a blueprint for this one :)

Consider that this would appear after the following line:
```python
log.info("Generating custom_slurm_settings_include_file_%s.conf", include_target)
```

Reply:
Addressed.